future/src/ct/ct_export2html.cc

```diff
@@ -345,11 +345,23 @@ Glib::ustring CtExport2Html::_get_table_html(CtTable* table)
 }
 
 // Get rich text from syntax highlighted code node
-Glib::ustring CtExport2Html::_html_get_from_code_buffer(Glib::RefPtr<Gsv::Buffer> code_buffer, int sel_start, int sel_end)
+Glib::ustring CtExport2Html::_html_get_from_code_buffer(const Glib::RefPtr<Gsv::Buffer>& code_buffer, int sel_start, int sel_end, const std::string& syntax_highlighting)
 {
     Gtk::TextIter curr_iter = sel_start >= 0 ? code_buffer->get_iter_at_offset(sel_start) : code_buffer->begin();
     code_buffer->ensure_highlight(curr_iter, code_buffer->end());
-    Glib::ustring html_text = "";
+    if (_pCtMainWin->get_ct_config()->usePandoc && CtPandoc::supports_syntax(syntax_highlighting)) {
+        if (CtPandoc::has_pandoc()) {
+            Glib::ustring      input_txt(curr_iter, code_buffer->end());
```

Comment:
not a `code_buffer->end()` but `sel_end`
function is used to get html from selected text